src/assisted_test_infra/test_infra/controllers/node_controllers/nutanix_controller.py

```diff
@@ -0,0 +1,241 @@
+import ipaddress
+from builtins import list
+from typing import Any, Callable, Dict, List, Tuple, Union
+
+import libvirt
+from nutanix_api import NutanixApiClient, NutanixCluster, NutanixVM
+from nutanix_api.nutanix_vm import PowerState, VMBootDevices
+
+from assisted_test_infra.test_infra import BaseClusterConfig
+from assisted_test_infra.test_infra.controllers.node_controllers.disk import Disk
+from assisted_test_infra.test_infra.controllers.node_controllers.node import Node
+from assisted_test_infra.test_infra.controllers.node_controllers.node_controller import NodeController
+from assisted_test_infra.test_infra.helper_classes.config.base_nutanix_config import BaseNutanixConfig
+from assisted_test_infra.test_infra.tools import terraform_utils
+from assisted_test_infra.test_infra.utils import TerraformControllerUtil, utils
+from service_client import log
+
+
+class NutanixController(NodeController):
+    _config: BaseNutanixConfig
+
+    def __init__(self, config: BaseNutanixConfig, cluster_config: BaseClusterConfig):
+        super().__init__(config, cluster_config)
+        self.cluster_name = cluster_config.cluster_name.get()
+        folder = TerraformControllerUtil.create_folder(self.cluster_name, platform=cluster_config.platform)
```

Comment:
Does Nutanix have folders?

Reply:
This is the local terraform folder. I think that Nutanix doesn't have folders.